engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/multiinstance/MultiInstanceBodyActivatedHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.handlers.multiinstance;
+
+import io.zeebe.engine.processor.workflow.BpmnStepContext;
+import io.zeebe.engine.processor.workflow.BpmnStepHandler;
+import io.zeebe.engine.processor.workflow.deployment.model.BpmnStep;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableMultiInstanceBody;
+import io.zeebe.engine.state.instance.VariablesState;
+import io.zeebe.msgpack.query.MsgPackQueryProcessor;
+import io.zeebe.protocol.impl.record.value.workflowinstance.WorkflowInstanceRecord;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import java.util.function.Function;
+import org.agrona.DirectBuffer;
+
+public class MultiInstanceBodyActivatedHandler extends AbstractMultiInstanceBodyHandler {
+
+  public MultiInstanceBodyActivatedHandler(Function<BpmnStep, BpmnStepHandler> innerHandlerLookup) {
+    super(WorkflowInstanceIntent.ELEMENT_COMPLETING, innerHandlerLookup);
+  }
+
+  @Override
+  protected boolean handleMultiInstanceBody(BpmnStepContext<ExecutableMultiInstanceBody> context) {
+    final VariablesState variablesState = context.getElementInstanceState().getVariablesState();
+
+    final MsgPackQueryProcessor.QueryResult result =
+        readInputCollectionVariable(context).getSingleResult();
+
+    final int createdInstances = result.readArray(item -> createInnerInstance(context, item));
+
+    final boolean transitionToCompleting = createdInstances == 0;
```

Comment:
Can we just return `createdInstances == 0` immediately? Meaning of it should be documented or clear from usage of the method itself (i.e. from the caller)